### PR TITLE
Fixes missing variant values in views-row rendering.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Fixes up missing variant information for views_row preprocess hook.

# Needed By (Date)
- ASAP por favor.

# Urgency
- Blocker for Earth launch.

# Steps to Test

1. Check out this branch
2. Navigate to the /earth-matters page
3. Edit a news item
4. Select a style
5. Review that style is maintained on /earth-matters
6. Change style
7. Review that style is maintained on /earth-matters and filtered pages.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/ui_patterns_variants/ui_patterns_variants.module [hunk 1]
```diff
@@ -144,10 +144,36 @@ function ui_patterns_variants_preprocess_views_view_field(&$variables) {
...
+  }
+
+  // Some times the view information is stored elsewhere.
+  if (isset($variables['options']['variants'])) {
+    $variants = $variables['options']['variants'];
+    foreach ($variants as $variant_name => $variant_values) {
+
+      // Check dynamic fist.
+      if (isset($variant_values['dynamic_value']) && $variant_values['dynamic_value'] !== "") {
+        $field_data = $row->_entity->get($variant_values['dynamic_value'])->getValue();
```
Comment (pookmish) [inline modules/ui_patterns_variants/ui_patterns_variants.module]: This might cause an issue if the dynamic value is actually some joined entity.

**Review by pookmish — CHANGES_REQUESTED** [review]

File: modules/ui_patterns_variants/ui_patterns_variants.module (hunk 1 above)
Comment (pookmish) [reply modules/ui_patterns_variants/ui_patterns_variants.module]: Also this won't respect view rewrite/replacements.

Comment (sherakama) [reply modules/ui_patterns_variants/ui_patterns_variants.module]: When would we ever try to map a value to en entity reference? I thought it is required to map to a field with values.

Comment (sherakama) [reply modules/ui_patterns_variants/ui_patterns_variants.module]: > Also this won't respect view rewrite/replacements.

How so?

Comment (pookmish) [reply modules/ui_patterns_variants/ui_patterns_variants.module]: it gets the entity value straight from the node. not what comes out of the view.
`->_entity` is a Node object.

Comment (sherakama) [reply modules/ui_patterns_variants/ui_patterns_variants.module]: Oh right... hmm.

Comment (sherakama) [reply modules/ui_patterns_variants/ui_patterns_variants.module]: screen shot 2017-10-23 at 4 45 27 pm

I don't see where the raw value from the view is. Perhaps this is the wrong hook.

Comment (pookmish) [reply modules/ui_patterns_variants/ui_patterns_variants.module]: yeah i remember thats why i had to do the preprocess_field, and then work my way up the chain. i can take a look. i got some time today

Comment (sherakama) [reply modules/ui_patterns_variants/ui_patterns_variants.module]: Perhaps it only provides a raw value if the field has a re-write or a modifier of some sort? I'll have to test.

Comment (sherakama) [reply modules/ui_patterns_variants/ui_patterns_variants.module]: Nope, same results with modified fields.

**Comment by pookmish** [comment]
That works!

**Comment by sherakama** [comment]
woot!

**Pull request merged**